src/hooks/useCioAutocomplete.ts

```diff
@@ -14,24 +14,48 @@ import {
   getItemsForActiveSections,
   getSearchSuggestionFeatures,
   trackRecommendationView,
+  toKebabCase,
 } from '../utils';
 import useConsoleErrors from './useConsoleErrors';
 import useSections from './useSections';
 import useRecommendationsObserver from './useRecommendationsObserver';
 
 export const defaultSections: UserDefinedSection[] = [
   {
-    identifier: 'Search Suggestions',
+    indexSection: 'Search Suggestions',
     type: 'autocomplete',
   },
   {
-    identifier: 'Products',
+    indexSection: 'Products',
     type: 'autocomplete',
   },
 ];
 
 export type UseCioAutocompleteOptions = Omit<CioAutocompleteProps, 'children'>;
 
+const convertLegacyParametersAndAddDefaults = (sections: UserDefinedSection[]) =>
+  sections.map((config) => {
+    const { type } = config;
+
+    if (type === 'recommendations') {
+      if (config.identifier && !config.podId) {
+        return { ...config, podId: config.identifier };
+      }
+
+      if (!config.indexSection) {
+        return { ...config, indexSection: 'Products' };
+      }
+    }
+
+    if (type === 'autocomplete' || !type) {
+      if (config.identifier && !config.indexSection) {
+        return { ...config, indexSection: config.identifier };
```

Comment:
Maybe we can add `type: 'autocomplete'` here when `!type` instead of making it required

Context https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/112/files#r1436618676

Reply:
We can, I'm just not entirely sure how we would type it. Let's pair on it